### PR TITLE
Fix the changeset publish

### PR DESCRIPTION
**Pull request by collins-w**
- Add build commands to ensure that the lib folder exists before publishing

### CODE REVIEW

File: .github/actions/prepare/action.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ runs:
         node-version: 20.11.1
         cache: 'pnpm'
     - name: Install dependencies
-      run: |
-        pnpm install-deps
+      run: pnpm run install-deps
+      shell: bash
+    - name: Build artifacts
+      run: pnpm run build --skip-nx-cache
```
Comment (tirumerla) [inline .github/actions/prepare/action.yml:24]: Can you also add test step as well to verify all tests pass?

File: .github/workflows/version-or-publish.yml [hunk 2]
```diff
@@ -46,23 +46,56 @@ jobs:
...
+
+      - name: Unpack the zipped artifact
+        run: |
+          set -euo pipefail
+          cd "${{ steps.temp-dir.outputs.path }}"
+          tar -xzvf "${{ needs.provenance.outputs.package-name }}" -C $GITHUB_WORKSPACE
+          cd "$GITHUB_WORKSPACE/package"
+          pnpm install --no-frozen-lockfile
+          pnpm nx-test-skip-cache
+          cd ..
```
Comment (tirumerla) [inline .github/workflows/version-or-publish.yml:88]: Not sure how changeset can detect its a git repo without `cd package/; git init` for creating git tags. Not sure if you tested this yet by using dry run

File: .github/workflows/version-or-publish.yml [hunk 3]
```diff
@@ -46,23 +46,56 @@ jobs:
...
+          path: "${{ steps.temp-dir.outputs.path }}"
+          sha256: ${{ needs.provenance.outputs.provenance-download-sha256 }}
+
+      - name: Unpack the zipped artifact
+        run: |
+          set -euo pipefail
+          cd "${{ steps.temp-dir.outputs.path }}"
+          tar -xzvf "${{ needs.provenance.outputs.package-name }}" -C $GITHUB_WORKSPACE
+          cd "$GITHUB_WORKSPACE/package"
+          pnpm install --no-frozen-lockfile
```
Comment (tirumerla) [inline .github/workflows/version-or-publish.yml:86]: use `install-deps`

**Review by tirumerla — APPROVED** [review]
lgtm, thanks

**Review by MCarlomagno — APPROVED** [review]

**Pull request merged**